### PR TITLE
Add Lat/lon grid generation, road snapping, and multiple images per lat/lon functionality.

### PR DESCRIPTION
**Pull request by marict**
The idea here is that we will query a NxN rectangular grid of lat/lon coordinates, we can use the SnapToRoad google API to snap these points to a road. This allows us to sample at some granularity all the roads in the NxN rectangle. For lower resolutions we might lose queries on smaller roads.

**Key Changes:**

1. **Latitude/Longitude Grid Generation**:
   - A new file, `get_point_grid.py`, has been added which includes the `generate_grid` function. This function creates a grid of latitude and longitude points within specified boundaries and resolution settings.
   - The grid can handle both positive and negative step values and adjusts the step direction based on the input start and end values.

2. **Road Snapping**:
   - The `snap_to_roads` function within `get_point_grid.py` takes the generated grid points and snaps them to the nearest roads.

3. **Multiple Image Fetching**:
   - Modifications in `get_images.py` allow fetching multiple images for a given latitude and longitude. The number of images and the distribution of their headings can be specified. The `--num-images` parameter replaces the previous `--heading` parameter, now calculating multiple headings to capture images from different perspectives around the specified location.

4. **Enhanced Command Line Arguments**:
   - Both scripts now support enhanced command-line argument parsing 

5. **We should test this**

### CODE REVIEW

File: get_images.py [hunk 1]
```diff
@@ -95,27 +94,40 @@ def get_image(lat: float, lon: float, heading: int = 0, fov: int = 120, size: st
...
+            # Get remove dots from lat and lon
+            lat_str = str(lat).replace(".", "dot")
+            lon_str = str(lon).replace(".", "dot")
+            output_file = output_dir / f"streetview_{metadata['date']}_{lat_str}_{lon_str}_{heading}.jpg"
+            with open(output_file, "wb") as file:
+                file.write(image)
+            print(f"Image saved to {output_file}")
+        else:
+            print("Failed to download image.")
+            exit()
```
Comment (vcanaa) [inline get_images.py]: Maybe raise an exception, so the caller can decided to quit or keep going.

Comment (marict) [reply get_images.py]: added

File: get_point_grid.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        # Extract the latitude and longitude from each snapped point
+        return [(point['location']['latitude'], point['location']['longitude']) for point in snapped_points]
+    else:
+        print("API Request Failed:", response.text)
+        return []
+
+def parse_args() -> argparse.ArgumentParser:
+    """Parse command line arguments for the grid coordinates and API key, including grid resolution and maximum number of points."""
+    parser = argparse.ArgumentParser(description="Generate a grid of coordinates and snap them to the nearest roads.")
+    parser.add_argument("lat_start", type=float, help="Start latitude of the grid.")
```
Comment (vcanaa) [inline get_point_grid.py:42]: Curious if you have seen [click](https://click.palletsprojects.com/en/8.1.x/). I like it more now after seeing it at work.

**Review by vcanaa — APPROVED** [review]

File: get_point_grid.py (hunk 2 above)
Comment (marict) [reply get_point_grid.py:42]: I haven't! Pretty cool! Feels like over engineering for this use case but will definitely consider it in the future

**Pull request merged**